addons/sale/models/account_move.py

```diff
@@ -29,15 +29,17 @@ def unlink(self):
             downpayment_lines.unlink()
         return res
 
-    @api.depends('invoice_user_id')
+    @api.depends('invoice_user_id', 'partner_id')
     def _compute_team_id(self):
         sale_moves = self.filtered(lambda move: move.is_sale_document(include_receipts=True))
-        for ((user_id, company_id), moves) in groupby(
+        for ((user_id, company_id, partner_id), moves) in groupby(
             sale_moves,
-            key=lambda m: (m.invoice_user_id.id, m.company_id.id)
+            key=lambda m: (m.invoice_user_id.id, m.company_id.id, m.partner_id)
```

Comment:
```suggestion
            key=lambda m: (m.invoice_user_id.id, m.company_id.id, m.partner_id.team_id.id)
```

Reply:
Thanks for the suggestions! The changes have already been applied.